lib/db.c

```diff
@@ -74,12 +74,54 @@ static const uint32_t GRN_TABLE_PAT_KEY_CACHE_SIZE = 1 << 15;
       grn_string_get_normalized(ctx, nstr, &key, &key_size, NULL);\
       block\
       grn_obj_close(ctx, nstr);\
+    } else {\
+      char name[GRN_TABLE_MAX_KEY_SIZE];\
+      int name_size;\
+      name_size = grn_obj_name(ctx, (grn_obj *)(table), name, GRN_TABLE_MAX_KEY_SIZE);\
+      ERR(GRN_INVALID_ARGUMENT,\
+          "[key][normalize] failed to normalize: <%.*s>: <%.*s>",\
+          name_size, name,\
+          (int)key_size, (const char *)key);\
     }\
   } else {\
     block\
   }\
 } while (0)
 
+static grn_obj *
+grn_inspect_key_with_table_name(grn_ctx *ctx, grn_obj *buffer, grn_obj *table, const void *key, unsigned int key_size)
+{
+  grn_obj inspected;
+
+  GRN_TEXT_INIT(&inspected, 0);
+  grn_inspect_limited(ctx, &inspected, table);
```

Comment:
Do we need all table information?
Do we only need table type (`grn_table_type_inspect()`), table name (`grn_inspect_name()`) and key type (`grn_table_key_inspect()`)?

Reply:
I'll fix it.